engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnDecisionBehavior.java

```diff
@@ -215,11 +218,24 @@ private void writeDecisionEvaluationEvent(
             .setElementInstanceKey(context.getElementInstanceKey())
             .setElementId(context.getElementId());
 
+    final var decisionKeysByDecisionId =
+        decisionState
+            .findDecisionsByDecisionRequirementsKey(decision.getDecisionRequirementsKey())
+            .stream()
+            .collect(
+                Collectors.toMap(
+                    persistedDecision -> bufferAsString(persistedDecision.getDecisionId()),
+                    PersistedDecision::getDecisionKey));
```

Comment:
💭 I guess this approach is sub-optimal only when the evaluated decision and its required decisions are a small subset of all the decisions in a drg. But is fast for root decision evaluations, because the lookup uses a whilePrefix that finds all decisions instead of looking up each decision individually. Seems like a fine approach to me.

Reply:
True. The alternative would be API like `decisionState.findDecisionByDecisionRequirementsKeyAndDecisionId()`. The current approach should be fine for now because I don't see big DRGs in the community. We can change if needed :+1:

Reply:
True. The alternative would be API like `decisionState.findDecisionByDecisionRequirementsKeyAndDecisionId()`. The current approach should be fine for now because I don't see big DRGs in the community. We can change if needed :+1: